src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedMathUtils.java

```diff
@@ -136,4 +141,22 @@ public static double stdDev(final double ... values) {
         Utils.nonNull(values);
         return Math.sqrt(new Variance().evaluate(values));
     }
+
+    // given a list of options and a function for calculating their probabilities (these must sum to 1 over the whole list)
+    // randomly choose one option from the implied categorical distribution
+    public static <E> E randomSelect(final List<E> choices, final Function<E, Double> probabilityFunction) {
```

Comment:
First off, I won't pretend that I wrote it as I did for any reason other than ignorance of the library version!

The implementation of `EnumeratedDistribution` is not lazy -- you have to pre-calculate the pmf -- whereas the way I wrote it you only have to calculate until you make a selection.  For cases where your first state has a large weight, such as germline segmentation where the neutral state is WLOG the first, this is a significant optimization.  Then again, we only plan on using this in tests, and also my code assumes a normalized pmf, so perhaps this optimization is unnecessary.  Furthermore, in germline segmentation there are so few states and the expensive part of the transition matrix is cached after one calculation, so the speedup is strictly less than 3x. 

All in all, I think I agree with your suggestion.